bin/varnishd/cache/cache_vpi.c

```diff
@@ -94,6 +94,11 @@ VPI_vcl_select(VRT_CTX, VCL_VCL vcl)
 	if (IS_TOPREQ(req) && req->top->vcl0 != NULL)
 		return;		// Illegal, req-FSM will fail this later.
 
+	/* XXX VCL_Task* are somewhat duplicated to those in Req_Rollback called
+	 * from FSM for VCL_RET_VCL. Keeping them here to ensure there are no
+	 * tasks during calls to VCL_Rel / vcl_get
+	 */
+	VCL_TaskLeave(req->vcl, req->top->privs);
 	VCL_TaskLeave(req->vcl, req->privs);
 	if (IS_TOPREQ(req)) {
 		AN(req->top);
```

Comment:
```c
CHECK_OBJ_NOTNULL(req->top, REQTOP_MAGIC);
```

Reply:
I am going to add it, but I guess this is the kind of assertion we do not need that badly. If the `req` (and the `reqtop` is part of it) was broken, we would know much sooner.